pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -100,6 +100,68 @@ func (c *Controller) GetAppliedToGroupNum() int {
 	return c.ruleCache.GetAppliedToGroupNum()
 }
 
+// TODO: save namespace information in cache.
+func (c *Controller) GetNetworkPolicies() []v1beta1.NetworkPolicy {
+	var ret []v1beta1.NetworkPolicy
+	c.ruleCache.policySetLock.RLock()
+	defer c.ruleCache.policySetLock.RUnlock()
+	for uid := range c.ruleCache.policySet {
+		np := v1beta1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{Name: uid},
+		}
+		atgSet := map[string]struct{}{}
+		rules, _ := c.ruleCache.rules.ByIndex(policyIndex, uid)
+		for _, ruleObj := range rules {
+			rule := ruleObj.(*rule)
+			np.Rules = append(np.Rules, v1beta1.NetworkPolicyRule{
+				Direction: rule.Direction,
+				From:      rule.From,
+				To:        rule.To,
+				Services:  rule.Services,
+			})
+			for _, atg := range rule.AppliedToGroups {
+				atgSet[atg] = struct{}{}
+			}
+		}
+		for atg := range atgSet {
+			np.AppliedToGroups = append(np.AppliedToGroups, atg)
+		}
```

Comment:
AppliedToGroups is same for all rules of a policy,  no need to merge AppliedToGroups across rules owned by a policy.

Reply:
Good to know, updated.